src/entities/timeTrackings/timeTrackingsAdapter.ts

```diff
@@ -1,10 +1,37 @@
-import { createEntityAdapter } from '@reduxjs/toolkit';
+import { createEntityAdapter, createSelector } from '@reduxjs/toolkit';
 import { TimeTracking } from '@timetac/js-client-library';
 
 import { RootState } from '../../rootState';
+import { dateUtils } from '../../utils/dateUtils';
+import { generalSettingsSelectors } from '../generalSettings/generalSettingsAdapter';
 
 export const timeTrackingsAdapter = createEntityAdapter<TimeTracking>({
   sortComparer: (a, b) => a.id - b.id,
 });
 
 export const timetrackingSelectors = timeTrackingsAdapter.getSelectors((state: RootState) => state.core.entities.timetrackings);
+
+export const currentRunningTimeTrackingSelector: (
+  state: RootState
+) => TimeTracking | undefined = createSelector(timetrackingSelectors.selectAll, (timetrackings: TimeTracking[]) =>
+  timetrackings.find((timetracking) => timetracking.status === 1)
+);
+
+export const previousRunningTimeTrackingSelector: (state: RootState) => TimeTracking | undefined = createSelector(
+  timetrackingSelectors.selectAll,
+  (state) => generalSettingsSelectors.selectById(state, 'default_break_task_id')?.setting_value ?? '',
+  (timetrackings: TimeTracking[], breakTaskId: string) => {
+    const filteredTimetrackings = timetrackings.filter(
+      (timetracking) => timetracking.task_id !== parseInt(breakTaskId) && timetracking.end_time !== undefined
+    );
+    return filteredTimetrackings.length > 0
+      ? filteredTimetrackings.reduce((lastOne: TimeTracking, x: TimeTracking) =>
+          lastOne == null
+            ? x
+            : x.end_time && lastOne.end_time && dateUtils.parseServerDateTime(x.end_time) > dateUtils.parseServerDateTime(lastOne.end_time)
+            ? x
+            : lastOne
+        )
+      : undefined;
```

Comment:
We could rewrite it with a for loop and if's for more readability. And a bit improved performance I believe due to only doing one instead of two date parses:

```suggestion
    if(filteredTimetrackings.length === 0) return undefined;

    let latest = filteredTimetrackings[0];
    let latestDate = dateUtils.parseServerDateTime(latest.end_time as string);

    for (let i = 1; i < filteredTimetrackings.length; i++) {
      const endDate = dateUtils.parseServerDateTime(filteredTimetrackings[i].end_time as string);

      if(endDate > latestDate) {
        latest = filteredTimetrackings[i];
        latestDate = endDate;
      }
    }
    
    return latest;
```

Reply:
still, readability is not perfect maybe something like that would be better

`return filteredTimetrackings.sort((a,b) => a.end_time - b.end_time).pop();`

For sure performance is not perfect because we a sorting and picking, and not just picking...

But the real question is: What is the purpose of this selector? because I don't know why you want to select the last running timetracking of some random user.

Reply:
@benj56 suggestion was taken to replace solution with .reduce()

Reply:
we need selector as we have start & stop & pause buttons and we need to know which timetracking was running before for all possible combinations of user behavior scenarios